server/src/main/java/com/genymobile/scrcpy/wrappers/DisplayManager.java

```diff
@@ -1,28 +1,54 @@
 package com.genymobile.scrcpy.wrappers;
 
+import android.os.IInterface;
+
 import com.genymobile.scrcpy.DisplayInfo;
+import com.genymobile.scrcpy.Ln;
 import com.genymobile.scrcpy.Size;
 
-import android.os.IInterface;
-
 public final class DisplayManager {
     private final IInterface manager;
 
     public DisplayManager(IInterface manager) {
         this.manager = manager;
     }
 
-    public DisplayInfo getDisplayInfo() {
+    public DisplayInfo getDisplayInfo(int displayId) {
         try {
-            Object displayInfo = manager.getClass().getMethod("getDisplayInfo", int.class).invoke(manager, 0);
+            Object displayInfo = manager.getClass().getMethod("getDisplayInfo", int.class).invoke(manager, displayId);
             Class<?> cls = displayInfo.getClass();
             // width and height already take the rotation into account
             int width = cls.getDeclaredField("logicalWidth").getInt(displayInfo);
             int height = cls.getDeclaredField("logicalHeight").getInt(displayInfo);
             int rotation = cls.getDeclaredField("rotation").getInt(displayInfo);
-            return new DisplayInfo(new Size(width, height), rotation);
+            int layerStack = cls.getDeclaredField("layerStack").getInt(displayInfo);
+            int flags = cls.getDeclaredField("flags").getInt(displayInfo);
+            return new DisplayInfo(displayId, new Size(width, height), rotation, layerStack, flags);
+        } catch (Exception e) {
+            if (e instanceof NullPointerException) suggestFix(displayId, getDisplayIds());
```

Comment:
Which variable is supposed to be `null`? Please use an explicit check `!= null` rather than a `catch NullPointerException` (or equivalent).